INSTALL.md

```diff
@@ -96,50 +68,55 @@ apt-get install hip-nvcc
 
-## Verify your installation
-Run hipconfig (instructions below assume default installation path) :
-```shell
-/opt/rocm/bin/hipconfig --full
-```
-
-Compile and run the [square sample](https://github.com/ROCm-Developer-Tools/HIP/tree/master/samples/0_Intro/square). 
+## Build ROCclr
 
+ROCclr is defined on AMD platform that HIP use Radeon Open Compute Common Language Runtime (ROCclr), which is a virtual device interface that HIP runtimes interact with different backends.
+See https://github.com/ROCm-Developer-Tools/ROCclr
 
-# Building HIP from source
-HIP source code is available and the project can be built from source on the HCC platform. 
-
-1. Follow the above steps to install and validate the binary packages.
-2. Download HIP source code (from the [GitHub repot](https://github.com/ROCm-Developer-Tools/HIP).)
-3. Install HIP build-time dependencies using ```sudo apt-get install libelf-dev```.
-4. Build and install HIP (This is the simple version assuming default paths ; see below for additional options.)
+```
+git clone -b roc-3.5.x https://github.com/ROCm-Developer-Tools/ROCclr.git
+export ROCCLR_DIR="$(readlink -f ROCclr)"
+git clone -b roc-3.5.x https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime.git
+export OPENCL_DIR="$(readlink -f ROCm-OpenCL-Runtime)"
+cd "$ROCCLR_DIR"
+mkdir -p build;cd build
+cmake -DOPENCL_DIR="$OPENCL_DIR" -DCMAKE_INSTALL_PREFIX=</where/to/install/ROCclr> ..
+make
+sudo make install
+```
 
-By default, HIP uses HCC to compile programs. To use HIP-Clang, add -DHIP_COMPILER=clang to cmake command line.
+## Build HIP
 
 ```
-cd HIP
-mkdir build
-cd build
-cmake .. 
+git clone -b roc-3.5.x https://github.com/ROCm-Developer-Tools/HIP.git
+export HIP_DIR="$(readlink -f HIP)"
+cd "$HIP_DIR"
+mkdir -p build; cd build
+cmake -DCMAKE_BUILD_TYPE=Release -DHIP_COMPILER=clang -DHIP_PLATFORM=rocclr -DROCclr_DIR="$ROCCLR_DIR" -DLIBROCclr_STATIC_DIR="$ROCCLR_DIR/build" -DCMAKE_INSTALL_PREFIX=</where/to/install/hip> ..
 make
-make install
+sudo make install
 ```
 
-* Default paths:
-  * By default cmake looks for hcc in /opt/rocm/hcc (can be overridden by setting ```-DHCC_HOME=/path/to/hcc``` in the cmake step).*
-  * By default cmake looks for HSA in /opt/rocm/hsa (can be overridden by setting ```-DHSA_PATH=/path/to/hsa``` in the cmake step).*
-  * By default cmake installs HIP to /opt/rocm/hip (can be overridden by setting ```-DCMAKE_INSTALL_PREFIX=/where/to/install/hip``` in the cmake step).*
+## Default paths and environment variables
+
+   * By default HIP looks for HSA in /opt/rocm/hsa (can be overridden by setting HSA_PATH environment variable).
+   * By default HIP is installed into /opt/rocm/hip (can be overridden by setting HIP_PATH environment variable).
+   * By default HIP looks for clang in /opt/rocm/llvm/bin (can be overridden by setting HIP_CLANG_PATH environment variable)
+   * By default HIP looks for device library in /opt/rocm/lib (can be overriden by setting DEVICE_LIB_PATH environment variable).
```

Comment:
By default HIP looks for:
* `HSA` in `/opt/rocm/hsa` (can be overridden by HSA_PATH);
* `clang` in `/opt/rocm/llvm/bin` (can be overridden by HIP_CLANG_PATH);
* `Device Libraries` in `/opt/rocm/lib` (can be overriden by DEVICE_LIB_PATH).

By default HIP is installed into `/opt/rocm/hip` (can be overridden by HIP_PATH).